### ISSUE TITLE
Error deploying from Github - "\MSBuild\14.0\Bin\MSBuild.exe was unexpected at this time."

### ISSUE THREAD

**Issue by natedahl32** [projectkudu/kudu#1689]
I've been deploying to my Azure website with Kudu for 4 months now with no issues. I deployed today at 11:35am CST and it completed successfully. I then did another deploy at around 2:35pm CST and started getting the error "\MSBuild\14.0\Bin\MSBuild.exe was unexpected at this time.". The deployment log contains the following:

Command: deploy.cmd
\MSBuild\14.0\Bin\MSBuild.exe was unexpected at this time.
Handling .NET Web Application deployment.

I added an echo statement to the deploy.cmd to see what the %MSBUILD_PATH% variable was set to and it is outputting:

D:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe

Anything happen today sometime after 11:35am CST that would have caused this error? Nothing has changed in my deploy.cmd file for the past 4 months.

**Comment by ahmelsayed** [comment]
we changed the default msbuild used, the one we set in `MSBUILD_PATH` to MSBuild 14.0 in 6eb4ac9d65b4293d820f2c5a13d8c0117ab38553 and I think it went to production yesterday at some point. If your kudu site was already running before that then it's possible you were still using older kudu at 11:35am and switched to the newer one later today. You can verify from the deployment logs for the deployment you did on 11:35.

That's said, you should be able to easily work around that issue by surrounding  `%MSBUILD_PATH%` with double quotes. Sorry about breaking your deployment script!

**Comment by natedahl32** [comment]
Thanks for the response! Helped point me in the right direction.

I had tried that prior to posting but it was still throwing the error. After your response I went back and double checked that I didn't miss a spot where I wasn't surrounding the build path with double quotes. After further inspection I realized I was using MSBuild on another project where I wasn't preceding the line with `call :ExecuteCmd`. After adding that to the line that was missing it my error went away. I'm guessing whatever change was made now requires that when calling into MSBuild?
